routes/ecom/modules/calculate-shipping.js

```diff
@@ -115,7 +115,7 @@ module.exports = appSdk => {
     }
 
     // optional params to Correios services
-    if (params.subtotal) {
+    if (params.subtotal && !config.no_declare_value) {

```

Comment:
`params.subtotal` é opcional, e se não for definido você ainda vai mandar o valor declarado porque ele será calculado pelos itens, o certo seria você verificar a configuração aqui na verdade: https://github.com/ecomplus/app-correios/blob/91d597456452945c057dca1ef76cf9056405d224/routes/ecom/modules/calculate-shipping.js#L474